[MRG + 1] MEP12 of showcase's examples + other folders.

NelleV opened the pull request:
Small MEP12-ifying showcase.

refs #7206

NelleV commented on examples/images_contours_and_fields/interpolation_none_vs_nearest.py:
```diff
@@ -1,62 +0,0 @@
-"""
```
I removed this example is being broadly a duplicate of interpolation_methods.

anntzer replied:
It does point out an important difference between the "none" and "nearest" methods though.  This information needs to remain available somewhere.

NelleV replied:
You mean the text or the example itself?

anntzer replied:
Both of the text and the images IMO, though not necessarily as an example (eg. it could be a FAQ entry instead).

NelleV replied:
I think it is very unlikely that anyone goes to that example to understand the difference between None and nearest. I think the interpolation methods (accompanied by more text explanation) is more than enough to understand the difference.

jenshnielsen replied:
It does not have to be an example but I would be very sorry to see that figure go. I have on several occasions benefitted from having this figure to explain the difference.  A graphical explanation is so much more powerfull than text only. The [interpolation_methods.py](http://matplotlib.org/examples/images_contours_and_fields/interpolation_methods.html) example does nothing to explain this as `None` `none` and `nearest` are rendered identically

anntzer replied:
You mean `interpolation_methods.py`?  Right now it certainly doesn't contain all this information.

NelleV replied:
No, but too much information is also not useful. We need to drastically cut down the number of examples we have because we dilute the information too much. I can add more text on the interpolation_methods.py example but if I have to do this for all examples we'll never have a gallery or even documentation that is in any way useful.

NelleV replied:
I've added some detailed text to interpolation methods, mostly copy pasting from the example I removed.

I am stressing once again that we need to cut back the number of examples. Most are extremely poorly designed, with unclear message and can be merged into one single example. It drowns the information too much.

To be specific, we have more than 500 examples right now, many displaying several images. There is no way anyone can sort or process any kind of information from these. If it helps convince you, I can have a communication expert look at the examples and the text and help us with the process of cleaning up our gallery.

anntzer replied:
I completely agree with your assessment that we have too many examples and that many are poorly designed.  However this specific example is exactly addressing a very specific question (the apparent duplication of interpolation methods between "none" and "nothing").  Thus my suggestion to move it to a FAQ entry.

anntzer commented:
http://matplotlib.org/examples/pylab_examples/manual_axis.html seems worth removing as well.

story645 commented on examples/images_contours_and_fields/interpolation_methods.py:
```diff
@@ -1,18 +1,23 @@
+This example displays the difference between interpolation methods for imshow
+and matshow.
+
+If `intperpolation` is None, it defaults to the rc image.interpolation
```
typo: interpolation

story645 commented on examples/images_contours_and_fields/interpolation_methods.py:
```diff
@@ -1,18 +1,23 @@
+
+If `intperpolation` is None, it defaults to the rc image.interpolation
+parameter. If the interpolation is `none`, then no interpolation is performed
+for the Agg, ps and pdf backends. Other backends will defult to 'nearest'.
```
in the Agg, ps, and pdf
And "Other backends default to `nearest`"

story645 commented on examples/images_contours_and_fields/interpolation_methods.py:
```diff
@@ -1,18 +1,23 @@
-# from the docs:
+For the Agg, ps and pdf backends, interpolation = 'none' works well when a big
+image is scaled down, while interpolation = 'nearest' works well when a small
+image is blown up.
```
large image is scaled down, small image is scaled up. May as well keep the language parallel.

story645 commented on examples/shapes_and_collections/scatter_demo.py:
```diff
@@ -1,5 +1,9 @@
 """
-Simple demo of a scatter plot.
+=============
+Scatter plots
```
Scatter plot

NelleV commented:
I've fixed the typos in the text I added to the interpolation_methods example. That text clearly needs more work, but I think it is out of the scope of this PR.

anntzer commented:
Thanks, merging and opening a separate issue regarding interpolation_none_vs_nearest.
